pkg/daemon/route/route.go

```diff
@@ -280,51 +295,75 @@ func (m *Manager) SyncRoutes() error {
 		}
 	}
 
-	for _, info := range m.overlaySubnetInfoMap {
+	var underlaySubnetInfoMap SubnetInfoMap
+
+	if len(m.remoteUnderlaySubnetInfoMap) == 0 {
+		// ignore remote underlay subnets
+		underlaySubnetInfoMap = m.localUnderlaySubnetInfoMap
+	} else {
+		// consider both local and remote underlay subnets
+		underlaySubnetInfoMap = make(map[string]*SubnetInfo, len(m.remoteUnderlaySubnetInfoMap)+len(m.localUnderlaySubnetInfoMap))
+		for cidr, info := range m.localUnderlaySubnetInfoMap {
+			underlaySubnetInfoMap[cidr] = info
+		}
+		for cidr, info := range m.remoteUnderlaySubnetInfoMap {
+			underlaySubnetInfoMap[cidr] = info
+		}
+	}
+
+	for _, info := range m.localOverlaySubnetInfoMap {
 		// Append overlay from pod subnet rules which don't exist and adapter subnet configuration
 		if err := ensureFromPodSubnetRuleAndRoutes(info.forwardNodeIfName, info.cidr,
 			info.gateway, info.autoNatOutgoing, true, m.family,
-			m.underlaySubnetInfoMap, underlayExcludeIPBlockMap); err != nil {
+			underlaySubnetInfoMap, localUnderlayExcludeIPBlockMap, remoteUnderlayExcludeIPBlockMap); err != nil {
```

Comment:
Why not considering both `localUnderlayExcludeIPBlockMap` and `remoteUnderlayExcludeIPBlockMap` as underlay excluded ip blocks? just like local and remote underlay subnets.

Then you might not need to make any change in pkg/daemon/route/utils.go.